src/org/labkey/test/components/ui/EntityBulkUpdateDialog.java

```diff
@@ -104,6 +104,25 @@ public String getNumericField(String fieldKey)
         return elementCache().numericInput(fieldKey).get();
     }
 
+    public EntityBulkUpdateDialog setDateField(String fieldKey, String dateString)
+    {
+        Input input = elementCache().textInput(fieldKey);
+        elementCache().title.click();
+        getWrapper().waitFor(()-> !isDatePickerExpanded(), 1500);
+        input.set(dateString);
+        return this;
+    }
+
+    private boolean isDatePickerExpanded()
+    {
+        return Locator.tagWithClass("div", "react-datepicker-popper").existsIn(this);
+    }
```

Comment:
Consider adding a `ReactDateInput` that handles waiting for the date picker popup to disappear after setting.

Reply:
Actually, is this a different field type than in `EntityBulkInsertDialog.setDateField`? They do different things so I'm wondering if one or the other is an earlier implementation or if they are actually different components.

Reply:
I wasn't aware that we had a component for a dateInput?
I did this in this way because I don't think insert or update dialogs have hit the case where they set the date field and then after need to interact with a following field (that would be obscured by the popup)

My thought is that a) we may want an abstract base class for the two dialogs, (they share a lot) and we will eventually want a wrapper to do dateInputs properly.  This method works... until we have a better dateInput handler

Reply:
I'll assume for now you meant that I should create a ReactDateInput that extends input, and handles the popup

Reply:
Unfortunately, it's hard to close the datepicker if you aren't either clicking in a safe spot outside of it, or selecting a time from the time-list.  I've got an attempt at a basic component that works so far in the use cases I have for it.